### PR TITLE
Cnft1-815: 508 Issue Duplicate Data (Phone Number)

### PR DESCRIPTION
**Pull request by Emelias-Enquizit**
**BEFORE:**

```json
{
  "data": {
    "findPatientsByFilter": {
      "total": 1,
      "content": [
        {
          "id": "10000001",
          "firstNm": "Surma",
          "middleNm": "J",
          "lastNm": "Singh",
          "addTime": "2015-09-22T14:40:46.137Z",
          "birthTime": "1990-01-01T00:00:00Z",
          "birthGenderCd": null,
          "currSexCd": "M",
          "ssn": null,
          "ethnicityGroupCd": null,
          "nbsEntity": {
            "entityLocatorParticipations": [
              {
                "classCd": "PST",
                "locator": {
                  "phoneNbrTxt": null,
                  "emailAddress": null,
                  "streetAddr1": "123 Main St."
                }
              },
              {
                "classCd": "TELE",
                "locator": {
                  "phoneNbrTxt": "232-322-2222",
                  "emailAddress": null,
                  "streetAddr1": null
                }
              },
              {
                "classCd": "TELE",
                "locator": {
                  "phoneNbrTxt": "456-232-3222",
                  "emailAddress": null,
                  "streetAddr1": null
                }
              },
              {
                "classCd": "TELE",
                "locator": {
                  "phoneNbrTxt": "232-322-2222",
                  "emailAddress": null,
                  "streetAddr1": null
                }
              },
              {
                "classCd": "TELE",
                "locator": {
                  "phoneNbrTxt": null,
                  "emailAddress": "fdsfs@dsds.com",
                  "streetAddr1": null
                }
              }
            ]
          }
        }
      ]
    }
  }
}

```

**AFTER:**

```json
{
  "data": {
    "findPatientsByFilter": {
      "total": 1,
      "content": [
        {
          "id": "10000001",
          "firstNm": "Surma",
          "middleNm": "J",
          "lastNm": "Singh",
          "addTime": "2015-09-22T14:40:46.137Z",
          "birthTime": "1990-01-01T00:00:00Z",
          "birthGenderCd": null,
          "currSexCd": "M",
          "ssn": null,
          "ethnicityGroupCd": null,
          "nbsEntity": {
            "entityLocatorParticipations": [
              {
                "classCd": "PST",
                "locator": {
                  "phoneNbrTxt": null,
                  "emailAddress": null,
                  "streetAddr1": "123 Main St."
                }
              },
              {
                "classCd": "TELE",
                "locator": {
                  "phoneNbrTxt": "232-322-2222",
                  "emailAddress": null,
                  "streetAddr1": null
                }
              },
              {
                "classCd": "TELE",
                "locator": {
                  "phoneNbrTxt": "456-232-3222",
                  "emailAddress": null,
                  "streetAddr1": null
                }
              },
              {
                "classCd": "TELE",
                "locator": {
                  "phoneNbrTxt": null,
                  "emailAddress": "fdsfs@dsds.com",
                  "streetAddr1": null
                }
              }
            ]
          }
        }
      ]
    }
```

### CODE REVIEW

File: apps/modernization-api/src/test/resources/features/patient/profile/PatientProfilePhoneChange.feature [hunk 1]
```diff
@@ -41,3 +41,10 @@ Feature: Patient Profile Phone Changes
     And I have the authorities: "FIND-PATIENT" for the jurisdiction: "ALL" and program area: "STD"
     Then I am unable to remove a patient's phone
     And a patient event is not emitted
+    
+   Scenario: I can update a patient's existing phone
```
Comment (adamloup-enquizit) [inline apps/modernization-api/src/test/resources/features/patient/profile/PatientProfilePhoneChange.feature:45]: These are tests for changing adding, updating, and removing Phone Numbers for an existing patient.  I think this story is meant for removing duplicates in Patient Search Results.

**Review by adamloup-enquizit — CHANGES_REQUESTED** [review]
One of the tests are failing due to a call to `this.input.active()` when not adding a new Patient.  Since a phone number is being added to an existing patient you might want to add a method in the `PatientMother` to add a phone number with a specific number value in order to detect duplicates.   

The Patient Search BDD tests are in the PatientSearch.feature` file.  These tests can be a huge hassle to set up.  If you want you could extract the `distinctNumbers` method into it's own class and test is in isolation.

**Comment by Emelias-Enquizit** [comment]
> One of the tests are failing due to a call to `this.input.active()` when not adding a new Patient. Since a phone number is being added to an existing patient you might want to add a method in the `PatientMother` to add a phone number with a specific number value in order to detect duplicates.
> 
> The Patient Search BDD tests are in the PatientSearch.feature`file. These tests can be a huge hassle to set up. If you want you could extract the`distinctNumbers` method into it's own class and test is in isolation.

HI @adamloup-enquizit  sure I will look into and yeah I will take the route of separating thedistinctNumbers and testing it in isolation

File: apps/modernization-api/src/test/resources/features/patient/profile/PatientProfilePhoneChange.feature (hunk 1 above)
Comment (Emelias-Enquizit) [reply apps/modernization-api/src/test/resources/features/patient/profile/PatientProfilePhoneChange.feature:45]: sure I will test it in isolation as mentioned in the quote reply.

Comment (Emelias-Enquizit) [reply apps/modernization-api/src/test/resources/features/patient/profile/PatientProfilePhoneChange.feature]: @adamloup-enquizit  updated.

File: apps/modernization-api/src/test/java/gov/cdc/nbs/patient/profile/phone/PatientProfilePhoneSteps.java [hunk 2]
```diff
@@ -34,11 +42,33 @@ public class PatientProfilePhoneSteps {
...
         mother.withPhone(patients.one());
+       
     }
+    
+	@Given("the patient has multiple phones")
+	public void the_patient_has_multiple_phones() {
+		persons = new ArrayList<>();
+		mother.withPhone(patients.one());
+
+		PatientInput.PhoneNumber phoneNumber = new PatientInput.PhoneNumber();
```
[the remark this answers is not in the document]
Comment (Emelias-Enquizit) [reply apps/modernization-api/src/test/java/gov/cdc/nbs/patient/profile/phone/PatientProfilePhoneSteps.java]: @adamloup-enquizit  updated

**Review by adamloup-enquizit — APPROVED** [review]
👍 💯

**Pull request merged**